### DISCUSSION TITLE
Changed shape on virtual character

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#81] [Q&A]
I am curious what happens to the original shape on the virtual character, after we use `character.SetShape()` on it. Who becomes the owner of it? Do we need to manually destroy the original shape, if it is no longer needed?

**Comment by jrouwe** [comment]
Shapes are reference counted, and the character keeps a reference. So if you replace the shape and don't keep a reference yourself (through e.g. CharacterVirtualSettings) then it will get deleted. I may have to expose the AddRef/Release interfaces so that you can keep a reference without Jolt's automatic reference counting smart pointer ```Ref<T>``` (which I don't think I can expose to JS).

**Reply by jrouwe** [reply]
I added the AddRef/Release functions and also updated the example. Note that you can also call GetRefCount on the objects to see what their current ref counts are for debugging purposes.

**Reply by LeXXik** [reply]
Thank you! This is very handy!
